File: src/main/java/guru/xgm/language/jass/openapi/actionSystem/convert/jass2angelscript/Jass2AngelScriptAction.java
package guru.xgm.language.jass.openapi.actionSystem.convert.jass2angelscript;

import guru.xgm.language.angelscript.openapi.fileTypes.AngelScriptFileType;
import guru.xgm.language.jass.openapi.actionSystem.convert.Jass2AnyAction;
import guru.xgm.language.jass.openapi.actionSystem.convert.Jass2AnyVisitor;
import org.jetbrains.annotations.NotNull;

public class Jass2AngelScriptAction extends Jass2AnyAction {

    public Jass2AngelScriptAction() {
        super();
    }

    @Override
    public @NotNull Jass2AnyVisitor getVisitor() {
        return new Jass2AngelScriptVisitor();
    }

    @Override
    public @NotNull String getTargetExtension() {
        return AngelScriptFileType.EXTENSION;
    }
}

File: src/main/kotlin/guru/xgm/language/jass/openapi/actionSystem/convert/jass2angelscript/Jass2AngelScriptAction.kt
package guru.xgm.language.jass.openapi.actionSystem.convert.jass2angelscript

import guru.xgm.language.angelscript.openapi.fileTypes.AngelScriptFileType
import guru.xgm.language.jass.openapi.actionSystem.convert.Jass2AnyAction
import guru.xgm.language.jass.openapi.actionSystem.convert.Jass2AnyVisitor

class Jass2AngelScriptAction : Jass2AnyAction() {
    override val visitor: Jass2AnyVisitor
        get() = Jass2AngelScriptVisitor()

    override val targetExtension: String
        get() = AngelScriptFileType.EXTENSION
}
